### PR TITLE
Adomerge packages

### PR DESCRIPTION
**Pull request by iwangjintian**
Helm package reorg to be extendable 
Adding TLS+SAT deployment
Added ginkgo scenario test from ADO

### CODE REVIEW

File: packages/helm/symphony-api.yaml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+            valueFrom:
+              fieldRef:
+                fieldPath: spec.serviceAccountName
+          - name: SYMPHONY_CONTROLLER_SERVICE_ACCOUNT_NAME
+            value: '{{ include "symphony.fullname" . }}-controller-manager'
+          {{ toYaml .Values.Symphony.api.env | indent 10 }}
+        envFrom:
+          - secretRef:
+              name: {{ include "symphony.fullname" . }}-auth
+          {{ if .Values.Symphony.api.proxySettings.isProxyEnabled }}
```
Comment (msftcoderdjw) [inline packages/helm/symphony-api.yaml:57]: Glad to see we have outbound proxy feature in symphony-api, please review all related PRs in ADO repo and make sure we merge all of them.

File: packages/helm/symphony/azure/metadata.json [hunk 2]
```diff
@@ -1,37 +1,37 @@
 {
```
Comment (msftcoderdjw) [inline packages/helm/symphony/azure/metadata.json:1]: No need to keep this file in OSS repo, it is only needed in Azure. We will patch them when generating extension.

File: packages/helm/symphony/crds/cert-manager.crdsv1.11.yaml
Comment (msftcoderdjw) [inline packages/helm/symphony/crds/cert-manager.crdsv1.11.yaml:1]: Do we still need to keep cert-manager.crds 1.11 in repo? I saw you bump the base yaml (cert-manager.crds.yaml) to 1.13.

File: packages/helm/symphony/files/azure/prepare-namespaces.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#!/bin/env bash
```
Comment (msftcoderdjw) [inline packages/helm/symphony/files/azure/prepare-namespaces.sh:1]: Where do we use prepare-namepsaces.sh? If this file is Azure only, please keep in extension repo and patch them in build.

File: packages/helm/symphony/files/trace-middleware.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
+{
+    "type": "middleware.http.tracing",
+    "properties": {
+      "serviceName": "aio-orc-api",
```
Comment (msftcoderdjw) [inline packages/helm/symphony/files/trace-middleware.json:4]: serviceName is `symphony-api`

File: packages/helm/symphony/values.yaml [hunk 5]
```diff
@@ -31,4 +31,14 @@ parent:
   username: admin
   password:
 siteId: hq
-imagePrivateRegistryUrl: ghcr.io
\ No newline at end of file
+imagePrivateRegistryUrl: ghcr.io
+api:
```
Comment (msftcoderdjw) [inline packages/helm/symphony/values.yaml:35]: You need to change `enabled` to `isEnabled` under global.azure.identity.

File: packages/testutils/conditions/jq/README.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+  - [func \(j \*JqCondition\) IsSatisfiedBy\(c context.Context, resource interface\{\}\) error](<#JqCondition.IsSatisfiedBy>)
+- [type Option](<#Option>)
+  - [func WithCustomMatcher\(matcher func\(ctx context.Context, value, root interface\{\}, log logger.Logger\) error\) Option](<#WithCustomMatcher>)
+  - [func WithDescription\(description string\) Option](<#WithDescription>)
+  - [func WithLogger\(log func\(format string, args ...interface\{\}\)\) Option](<#WithLogger>)
+  - [func WithValue\(value interface\{\}\) Option](<#WithValue>)
+
+
+<a name="JqCondition"></a>
+## type [JqCondition](<https://dev.azure.com/msazure/One/_git/symphony?path=packages%2Ftestutils%2Fconditions%2Fjq%2Fjq.go&version=GBmain&lineStyle=plain&line=17&lineEnd=25&lineStartColumn=2&lineEndColumn=3>)
```
Comment (msftcoderdjw) [inline packages/testutils/conditions/jq/README.md:26]: The link is still pointing to ADO repo, please fix/regenerate them all.

**Review by msftcoderdjw — CHANGES_REQUESTED** [review]

File: .github/workflows/suite.yml [hunk 7]
```diff
@@ -92,14 +92,21 @@ jobs:
...
       run: |
         cd test/integration/scenarios/06.ado/ 
-        ginkgo --cover --junit-report=junit-suite-tests.xml -r 
+        ginkgo --cover --junit-report=junit-suite-tests.xml -r > suite_test.log
       continue-on-error: true    
+
+    - name: Dump SymphonyLogs For ginkgo suite tests
+      run: |
+        cd test/localenv/
+        mage DumpSymphonyLogsForTest ginkgosuite
```
Comment (msftcoderdjw) [inline .github/workflows/suite.yml:101]: continue-on-error: true

File: go.work.bk
Comment (Haishi2016) [inline go.work.bk:1]: Do we need this file?

File: api/Dockerfile [hunk 8]
```diff
@@ -38,4 +38,4 @@ EXPOSE 8080
 EXPOSE 8081
 ENV LOG_LEVEL Debug
 # ENV CONFIG /symphony-api.json
-CMD exec /symphony-api -c $CONFIG -l $LOG_LEVEL
\ No newline at end of file
+CMD sh -c 'if [ -f /etc/pki/ca-trust/source/anchors/proxy-cert.crt ]; then update-ca-trust; fi && exec /symphony-api -c $CONFIG -l $LOG_LEVEL'
```
Comment (Haishi2016) [inline api/Dockerfile:41]: Can we log an issue to update the doc on how certificates are configured|? Similar to Azure proxy configuration.

File: packages/helm/symphony/files/symphony-api.json [hunk 9]
```diff
@@ -212,7 +212,7 @@
             "type": "managers.symphony.jobs",
             "properties": {
               "providers.state": "mem-state",
-              "baseUrl": "http://symphony-service:8080/v1alpha2/",
+              "baseUrl": "https://symphony-service:8081/v1alpha2/",
```
Comment (Haishi2016) [inline packages/helm/symphony/files/symphony-api.json:215]: If the port is configured by symphony.apiContainerPortHttp, it should be used here as well, right?

File: packages/helm/symphony/templates/identity/extension-identity.yaml [hunk 10]
```diff
@@ -1,4 +1,4 @@
-{{ if .Values.global.azure.identity.enabled -}}
+{{ if .Values.global.azure.identity.isEnabled -}}
 apiVersion: clusterconfig.azure.com/v1beta1
 kind: AzureExtensionIdentity
```
Comment (Haishi2016) [inline packages/helm/symphony/templates/identity/extension-identity.yaml:3]: When can we assume this kind exist?

File: go.work.bk
Comment (iwangjintian) [reply go.work.bk:1]: Yes. It's for unified dev experience.

File: api/Dockerfile (hunk 8 above)
Comment (iwangjintian) [reply api/Dockerfile:41]: https://github.com/eclipse-symphony/symphony/issues/239

File: packages/helm/symphony/templates/identity/extension-identity.yaml (hunk 10 above)
Comment (iwangjintian) [reply packages/helm/symphony/templates/identity/extension-identity.yaml:3]: this is for the extension msi

**Review by msftcoderdjw — APPROVED** [review]

**Review by msftcoderdjw — DISMISSED** [review]

**Review by msftcoderdjw — APPROVED** [review]

**Pull request merged**